src/test/scala-2.11/scala/tools/refactoring/tests/implementations/imports/OrganizeImportsWithMacrosTest.scala

```diff
@@ -0,0 +1,165 @@
+package scala.tools.refactoring
+package tests.implementations.imports
+
+import implementations.OrganizeImports.Dependencies
+
+class OrganizeImportsWithMacrosTest extends OrganizeImportsBaseTest {
+  private def organizeWithTypicalParams(pro: FileSet) = organizeCustomized()(pro)
+
+  private def organizeCustomized(
+      groupPkgs: List[String] = List("java", "scala", "org", "com"),
+      useWildcards: Set[String] = Set("scalaz", "scalaz.Scalaz"),
+      dependencies: Dependencies.Value = Dependencies.RemoveUnneeded,
+      organizeLocalImports: Boolean = true)(pro: FileSet) = new OrganizeImportsRefatoring(pro) {
+    val params = {
+      val groupImports = refactoring.GroupImports(groupPkgs)
+      val alwaysUseWildcards = refactoring.AlwaysUseWildcards(useWildcards)
+
+      new refactoring.RefactoringParameters(
+          options =
+            refactoring.ExpandImports ::
+            refactoring.PrependScalaPackage ::
+            alwaysUseWildcards ::
+            refactoring.SortImports ::
+            groupImports ::
+            Nil,
+          deps = dependencies,
+          organizeLocalImports = organizeLocalImports)
+    }
+  }.mkChanges
+
+  @Ignore("""Needs CLASSPATH set to project classpath to allow interactive global to see macro defined below.
+      Go to 'Environment' tab of launch configuration and add CLASSPATH variable set to {project_classpath:org.scala-refactoring.library}.""")
+  @Test
+  def shouldOrganizeImports() = new FileSet {
+    """
+    /*<-*/
+    package acme
+
+    import scala.tools.refactoring.tests.implementations.imports.OrganizeImportsWithMacros.FakeType
+    import scala.tools.refactoring.tests.implementations.imports.OrganizeImportsWithMacros.Macro
+
+    case class SomeCaseClass(x: Int)
+
+    object MacroConsumer {
+      import scala.tools.refactoring.tests.implementations.imports.OrganizeImportsWithMacros.Macro._
+
+      implicit val someClassCallMe = Macro.callMe[SomeCaseClass]
+    }
+
+    case class Bug2(x: SomeCaseClass)
+
+    object Bug2 {
+      import MacroConsumer._
+
+      implicit val bug2CallMe = Macro.callMe[Bug2]
+    }
+    """ isNotModified
+  } applyRefactoring organizeWithTypicalParams
+}
+
+object OrganizeImportsWithMacros {
+  import scala.language.experimental.macros
+  import scala.language.higherKinds
+  import scala.reflect.macros.blackbox.Context
+
+  trait FakeType[A]
+
+  object Macro {
+    implicit val fakeTypeInt = new FakeType[Int] {}
+    def callMe[A] = macro MacroTest.macroImpl[A, FakeType]
+  }
+
+  /**
+   * Implementation for the macro.
```

Comment:
Can you note here that the macro implementation is inspired by the play `Json.format` macro in order to reproduce a bug?